doc/dev/plugins-ordering.md

```diff
@@ -43,28 +54,34 @@ rollback or error notification are inserted into it.
 The resolver plugin requires this error list to do a proper rollback.
 Another use case is logging after a failure has happened.
 
+## Arrays of linked lists
+
+A disadvantage of using arrays is that they contain a fixed amount of slots. This means
+that when a certain amount of slots is full, it is no longer possible to add more.
+Therefore, the plugin arrays are structured as arrays of linked lists.
+
+Each plugin role is assigned its own slot in the array, which contains a linked list. Plugins
+fulfilling that role are added to the list. That way, much greater and more flexible plugin
+numbers can be added to a single role.
+
 ## Placements
 
 The ordering of plugins inside these three arrays is controlled by
 [elektra-contracts(7)](/doc/help/elektra-contracts.md).
-Each of the three arrays has ten slots. These slots have
+Each of the three arrays has one slot for each role. These slots have
```

Comment:
Do we even need the term "slot" anymore?

Reply:
No, which is why there is a warning at the top of this file.

> **Note** Even some of the docs changes in this PR, could be outdated. That's because some of it is still from the original #2969 by vLesk. Some parts have changed since and haven't been updated yet.